cluster-autoscaler/cloudprovider/mcm/mcm_manager.go

```diff
@@ -395,6 +400,25 @@ func (m *McmManager) GetMachineDeploymentSize(machinedeployment *MachineDeployme
 	return int64(md.Spec.Replicas), nil
 }
 
+func isRollingUpdateFinished(md *v1alpha1.MachineDeployment) bool {
+	found := false
+	for _, cond := range md.Status.Conditions {
+		switch {
+		case cond.Type == machineDeploymentProgressing && cond.Status == conditionTrue && cond.Reason == newISAvailableReason:
+			return true
+		case cond.Type == machineDeploymentProgressing:
+			found = true
+			break
+		}
+	}
+	if !found {
+		// no "Progressing" condition means the deployment has not undergone any rolling update yet
+		return true
+	}
+
+	return false
+}
```

Comment:
```go
	for _, cond := range md.Status.Conditions {
		switch {
		case cond.Type == machineDeploymentProgressing && cond.Status == conditionTrue && cond.Reason == newISAvailableReason:
			return true
		case cond.Type == machineDeploymentProgressing:
			return false
		}
	}
	// no "Progressing" condition means the deployment has not undergone any rolling update yet
	return true
}
```
We do not need `found` variable, right? We can also use `if` instead of `switch`.

Reply:
yes `found` variable is useless. will remove it.
using `switch` as i have seen upstream kubernetes generally preferring switch for long if statements.

Reply:
addressed in 92972a3f355a596b32a322113f8bd08f7fae95b8

Reply:
thanks